src/internal/common/idname/idname.go

```diff
@@ -0,0 +1,109 @@
+package idname
+
+import (
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+// Provider is a tuple containing an ID and a Name.  Names are
+// assumed to be human-displayable versions of system IDs.
+// Providers should always be populated, while a nil values is
+// likely an error.  Compliant structs should provide both a name
+// and an ID, never just one.  Values are not validated, so both
+// values being empty is an allowed conditions, but the assumption
+// is that downstream consumers will have problems as a result.
+type Provider interface {
+	// ID returns the canonical id of the thing, generated and
+	// usable  by whichever system has ownership of it.
+	ID() string
+	// the human-readable name of the thing.
+	Name() string
+}
+
+var _ Provider = &Is{}
+
+// Is provides an id-name tuple.
+type Is struct {
```

Comment:
I think this struct is not referenced outside this package and could be unexported as well?